src/test/java/com/desiato/puresynth/controllers/UserControllerTest.java

```diff
@@ -0,0 +1,153 @@
+package com.desiato.puresynth.controllers;
+
+import com.desiato.puresynth.BaseTest;
+import com.desiato.puresynth.models.User;
+import com.desiato.puresynth.repositories.UserRepository;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+public class UserControllerTest extends BaseTest {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    private static final Logger logger = LoggerFactory.getLogger(LoginControllerTest.class);
+
+    @Test
+    public void createUser_ShouldReturnCreatedUser() throws Exception {
+        String uniqueEmail = generateUniqueEmail();
+        String newUserJson = String.format("""
+        {
+            "email": "%s",
+            "password": "password123"
+        }
+        """, uniqueEmail);
+
+        mockMvc.perform(post("/api/user")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(newUserJson))
+                .andExpect(status().isCreated())
+                .andExpect(jsonPath("$.email").value(uniqueEmail))
+                .andDo(print());
+    }
+
+    @Test
+    public void deleteUser_ShouldDeleteUserAndReturnOk() throws Exception {
```

Comment:
This test should use `logout` endpoint. I don't think you need to create this as part of this PR. This PR is already big.